src/js/utils/rxjsUtils.ts

```diff
@@ -0,0 +1,30 @@
+import { timer } from "rxjs/observable/timer";
+
+import "rxjs/add/operator/delayWhen";
+import "rxjs/add/operator/scan";
+import { Scheduler } from "rxjs/Scheduler";
+import { Observable } from "rxjs/Observable";
+
+export function linearBackoff(
+  delay: number,
+  maxRetries: number = -1,
+  maxDelay?: number,
+  scheduler?: Scheduler
+): (errors: Observable<any>) => Observable<any> {
+  return (errors: Observable<any>) =>
+    errors
+      .scan((count, error) => {
+        if (maxRetries >= 0 && count >= maxRetries) {
+          throw error;
+        }
+
+        return count + 1;
+      }, 0)
+      .delayWhen(val =>
+        timer(
+          Math.min(val * delay, maxDelay || Number.MAX_SAFE_INTEGER),
+          undefined,
```

Comment:
unusual seeing `undefined` to be passed to skip the default  value 🤷‍♂️

Reply:
In RxJs they use `private period: number = -1;` as default which could be a better value as it is more like a pattern https://github.com/ReactiveX/rxjs/blob/b5ed59d8626376d16ac09aa98ac9c00835340fe3/src/observable/TimerObservable.ts#L82

Reply:
ok!